src/Arcus.EventGrid.WebApi.Security/EventGridSubscriptionValidationActionFilter.cs

```diff
@@ -0,0 +1,105 @@
+using Microsoft.Extensions.Primitives;
+
+namespace Arcus.EventGrid.WebApi.Security
+{
+    /// <summary>
+    /// Represents a filter to validate Azure Event Grid subscription events.
+    /// </summary>
+    internal class EventGridSubscriptionValidationActionFilter : IAsyncActionFilter
+    {
+        /// <summary>
+        /// Called asynchronously before the action, after model binding is complete.
+        /// </summary>
+        /// <param name="context">The <see cref="T:Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext" />.</param>
+        /// <param name="next">
+        /// The <see cref="T:Microsoft.AspNetCore.Mvc.Filters.ActionExecutionDelegate" />. Invoked to execute the next action filter or the action itself.
+        /// </param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that on completion indicates the filter has executed.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="context"/> or <paramref name="next"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="context"/> doesn't contain the necessary information to validate the Azure Event Grid subscription.
+        /// </exception>
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            Guard.NotNull(context, nameof(context), "Requires an action filter HTTP context to validate the Azure Event Grid subscription");
+            Guard.NotNull(next, nameof(next), "Requires an action delegate function to run the next action filter or the action itself when the Azure Event Grid subscription validation is complete");
+            Guard.For(() => context.HttpContext is null, new ArgumentException("Requires a HTTP context in the action filter context to validate the Azure Event Grid subscription", nameof(context)));
+            Guard.For(() => context.HttpContext.Request is null, new ArgumentException("Requires a HTTP context with a HTTP request in the action filter context to validate the Azure Event Grid subscription", nameof(context)));
+            Guard.For(() => context.HttpContext.Request.Headers is null, new ArgumentException("Requires a HTTP context with a HTTP request containing headers in the action filter context to validate the Azure Event Grid subscription", nameof(context)));
+            Guard.For(() => context.HttpContext.Request.Body is null, new ArgumentException("Requires a HTTP context with a HTTP request containing a body in the action filter context to validate the Azure Event Grid subscription", nameof(context)));
+            
+            ILogger logger = GetRegisteredLogger(context.HttpContext.RequestServices);
+            
+            // TODO: configurable header name/value
+            if (context.HttpContext.Request.Headers.TryGetValue("Aeg-Event-Type", out StringValues eventTypes)
```

Comment:
So this check is not CloudEvents compliant then?

Example: https://github.com/tomkerkhove/autoscaling-over-time-dashboard/blob/main/src/AzureAutoscalingToolbox.Samples.StatefulAppInstances/Functions/KubernetesAppScaledFunction.cs#L30-L33

We should add an item for this

Reply:
What was this again, this link? Does CloudEvent compliant mean that we verify this HTTP OPTIONS request?

Reply:
Yes

Reply:
The approach above is the EventGrid schema way, not CloudEvents way.

Reply:
Maybe for dedicated issue, this PR was only to move it as-is and use our secret store instead. The docs are also for me not clear enough to what this is doing, so we should maybe ask @SamVanhoutte for help on this, where this is used. But again, dedicated issue would be better.

Reply:
https://github.com/arcus-azure/arcus.eventgrid/issues/180